### PR TITLE
Add mini-lesson on destructuring

### PR DESCRIPTION
**Pull request by Srashtimittal**
Issue Number : #1220
Issue Fixed: Add mini-lesson on destructuring

### CODE REVIEW

File: javascript/Mini-lesson on destructuring [hunk 1]
```diff
@@ -0,0 +1,57 @@
+### Projected Time
```
Comment (alodahl) [inline javascript/Mini-lesson on destructuring]: Add a header title please

File: javascript/Mini-lesson on destructuring [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+About 25-30 minutes
+- 10 min for destructuring
+- 20 min for examples
+
+### Prerequisites
+
+Here are the topic that should be understood before this topic:
+- Javascript 7
```
Comment (alodahl) [inline javascript/Mini-lesson on destructuring]: Provide the link

File: javascript/Mini-lesson on destructuring [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+object or an array. With the destructuring syntax, you can extract smaller fragments from arrays and objects. Destructuring syntax can be
+used for variable declaration or variable assignment.
+
+### Why it is useful?
+
+There are two big advantages of using destructuring. The first one is that your code more readable. If you destructure an object at the
+top of a function or code block, it is clear to the reader what variables you are going to use.
+The second plus is performance. Destructuring encourages programmers to put object properties in local variables before using them,
+which can improve application performance. Especially if you are accessing those variables multiple times, maybe in a loop, it is more
+efficiënt if the variables are locally defined.
```
Comment (alodahl) [inline javascript/Mini-lesson on destructuring]:
```suggestion
efficient if the variables are locally defined.
```

File: javascript/Mini-lesson on destructuring [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+### Examples
+
+### Example 1 ->
```
Comment (alodahl) [inline javascript/Mini-lesson on destructuring]: Please use proper markdown for subheaders- in the case you want “####’ for each example.

File: javascript/Mini-lesson on destructuring [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+Destructuring an two element array into two named variables.
+
+### Code with destructuring :
+
+let { type, color } = car;
```
Comment (alodahl) [inline javascript/Mini-lesson on destructuring]: Be sure to indent lines if they are supposed to be under a section. Also, use backticks to denote code.
```suggestion
  `let { type, color } = car;`
```

File: javascript/Mini-lesson on destructuring [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+which can improve application performance. Especially if you are accessing those variables multiple times, maybe in a loop, it is more
+efficiënt if the variables are locally defined.
+
+### Examples
+
+### Example 1 ->
+
+Destructuring an two element array into two named variables.
+
+### Code with destructuring :
```
Comment (alodahl) [inline javascript/Mini-lesson on destructuring]: If this is part of example one, you either want ‘#####’ or double asterisks to make it bold. Google ‘markdown cheatsheet’ If you’re not sure what this means.

File: javascript/Mini-lesson on destructuring [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+### Code without destructuring :
+
+let username = req.body.username;
+let password = req.body.password;
+let email    = req.body.email;
+
+### Check for Understanding 
+
+Some ideas : Take a quiz, do an assignment or something else that helps them to test the understanding .
```
Comment (alodahl) [inline javascript/Mini-lesson on destructuring]: Please put an actual activity or discussion here.

File: javascript/javascript-4.md [hunk 8]
```diff
@@ -1,4 +1,4 @@
-# JavaScript 4
+### JavaScript 4
```
Comment (alodahl) [inline javascript/javascript-4.md]: ?

File: javascript/Mini-lesson on destructuring (hunk 1 above)
Comment (Srashtimittal) [reply javascript/Mini-lesson on destructuring]: Ok , I will add

**Comment by Srashtimittal** [comment]
@alodahl Mam I update the file with the changes. So, please view the file.

File: javascript/Mini-lesson on destructuring (hunk 2 above)
Comment (alodahl) [reply javascript/Mini-lesson on destructuring]: this should be linked to the JS7 outline

File: javascript/Mini-lesson on destructuring [hunk 9]
```diff
@@ -0,0 +1,60 @@
+## Projected Time
```
Comment (alodahl) [inline javascript/Mini-lesson on destructuring]: Please add a title

File: javascript/Mini-lesson on destructuring [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+**Code without destructuring**
+
+`let username = req.body.username;`
+`let password = req.body.password;`
+`let email    = req.body.email;`
+
+## Check for Understanding 
+
+Open the given below link and solve the questions related to destructuring that helps them to test the understanding .
+<https://stackoverflow.com/questions/tagged/destructuring>
```
Comment (alodahl) [inline javascript/Mini-lesson on destructuring]: please use proper linking syntax.  check out https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#links

**Comment by Srashtimittal** [comment]
@alodahl Mam I make all the changes again. So, please review the file.

**Pull request merged**